### ISSUE TITLE
Child tab of duplicate tab can appear in wrong position

### ISSUE THREAD

**Issue by lbmaian** [piroor/treestyletab#1438]
## Short description
Child tab of duplicate tab can appear in wrong position

## Steps to reproduce

 1. Start Firefox with clean profile.
 2. Install TST.
 3. Tree structure:
```
A
\- B (collapsed)
C
\- D
```
 4. Close C/D tree:
```
A (selected)
\- B (can be collapsed if "Except focus moving caused by current tab closing" is enabled)
```
 5. Duplicate A via either middle-click of reload button or menu item (note: if this can result in another bug #1437; if so, rinse and repeat until it creates it properly):
```
A
\- B
A-dup (selected)
```
 6. Create child of A-dup

## Expected result
```
A
\- B (collapsed)
A-dup (selected)
\- C
```

## Actual result
```
A
|- B (collapsed)
\- C (but actually child of A-dup)
A-dup (selected)
```

## Environment

 * Platform (OS): Windows 10 Pro, Version 1709, Build 16299.15
 * Version of Firefox: Developer Edition 57.0b5
 * Version (or revision) of Tree Style Tab: Latest nightly as of commit f65ea7b1a02ddcb5ead6eac7a2ef5d6f2dfd495c

**Comment by piroor** [comment]
I couldn't reproduce this problem for now. Recent changes may solve this problem. Could you try the latest development build?

**Comment by lbmaian** [comment]
That particular edge case isn't reproducing for me anymore, but the following related one is:

1. Empty session (Firefox starts with only home page)
2. Tree structure:
```
A
\- B (collapsed)
C (selected)
\- D
```
3. Duplicate tab A via menu item:
```
A
\- B (collapsed)
A-dup (selected)
C
\- D
```
4. Duplicate tab A via menu item again

## Expected
```
A
\- B (collapsed)
A-dup2 (selected)
A-dup
C
\- D
```

## Actual
Sometimes:
```
A
A-dup2
\- B (but still child of A)
A-dup
C
\- D (collapsed)
```

Sometimes:
```
A
A-dup2
A-dup
\-B (but still child of A
C
\- D (collapsed)
```

Sometimes at step 3 (same or similar bug to #1437):
```
A
|- A-dup2
\- B
C
\- D
```

And sometimes flip A-dup and A-dup2 tabs around.

**Comment by lbmaian** [comment]
Come to think of it, that example may just be a manifestation of bug #1436 

The most notable property of this particular bug (1438) is the child tab of the duplicated tab appearing _above_ the duplicated tab as a "child" of the previous tree. I can't reproduce that now.

**Comment by lbmaian** [comment]
Can no longer reproduce with fixes for #1437
